client/src/pages/Signup.js

```diff
@@ -0,0 +1,223 @@
+import React from "react";
+import {
+  Container,
+  Typography,
+  Box,
+  TextField,
+  Divider,
+  Button,
+} from "@material-ui/core";
+import Grid from "@material-ui/core/Grid";
+import { makeStyles } from "@material-ui/core/styles";
+import { Link } from "react-router-dom";
+
+import NavBar from "../components/Navbar";
+
+import { validateEmail } from "../util/validateEmail";
+
+const useStyles = makeStyles((theme) => ({
+  header: {
+    padding: "2rem",
+    borderBottom: "2px",
+    borderBottomColor: theme.primary,
+  },
+  loginLink: {
+    marginLeft: "2px",
+  },
+  subtitle: {
+    padding: "2rem",
+  },
+  divider: {
+    backgroundColor: theme.primary,
+    height: "2px",
+    width: "10%",
+    margin: "auto",
+  },
+  button: {
+    backgroundColor: theme.primary,
+    marginTop: "4rem",
+    marginBottom: "4rem",
+    color: theme.bgcolor,
+    height: "3rem",
+    width: "60%",
+  },
+}));
+
+function Signup(props) {
+  const classes = useStyles();
+
+  const [state, setState] = React.useState({
+    name: "",
+    invalidName: false,
+    email: "",
+    invalidEmail: false,
+    password: "",
+    invalidPassword: false,
+    confirm: "",
+    invalidConfirm: false,
+  });
+
+  // Name handlers
+  const handleUpdateName = (event) => {
+    setState({ ...state, name: event.target.value });
+  };
+
+  const handleBlurName = (event) => {
+    if (state.name.length < 3 || state.name.length > 64) {
+      setState({ ...state, invalidName: true });
+    } else {
+      setState({ ...state, invalidName: false });
```

Comment:
I would try to reduce the num times you call `setState`

maybe 
```
const isValidName= state.name.length < 3 || state.name.length > 64;
setState({ ...state, invalidName: isValidName });
```

Reply:
A much cleaner approach, thanks!